var/spack/repos/builtin/packages/mysql/package.py

```diff
@@ -105,8 +119,37 @@ def cmake_args(self):
             options.append('-DWITHOUT_SERVER:BOOL=ON')
         return options
 
+    def fix_dtrace_usr_bin_path(self, spack_env):
+        # dtrace may cause build to fail because it uses
+        # '/usr/bin/python' in the shebang. To work around that we copy
+        # the original script into a temporary folder, and change the
+        # shebang to '/usr/bin/env python'. Treatment adapted from that
+        # used in glib recipe per M. Culpo @b2822b258.
+        dtrace = which('dtrace').path
+        dtrace_copy_path = os.path.join(tempfile.mkdtemp(), 'dtrace-copy')
+        dtrace_copy = os.path.join(dtrace_copy_path, 'dtrace')
+        mkdirp(dtrace_copy_path)
+        copy(dtrace, dtrace_copy)
+        filter_file(
+            '^#!/usr/bin/python',
+            '#!/usr/bin/env python',
+            dtrace_copy
+        )
+        # To have our own copy of dtrace in PATH, we need to
+        # prepend to PATH the temporary folder where it resides.
+        spack_env.prepend_path('PATH', dtrace_copy_path)
+
     def setup_environment(self, spack_env, run_env):
         cxxstd = self.spec.variants['cxxstd'].value
         flag = getattr(self.compiler, 'cxx{0}_flag'.format(cxxstd))
         if flag:
             spack_env.append_flags('CXXFLAGS', flag)
+        if cxxstd != '98':
+            if int(cxxstd) > 11:
+                spack_env.append_flags('CXXFLAGS',
+                                       '-Wno-deprecated-declarations')
+            if int(cxxstd) > 14:
+                spack_env.append_flags('CXXFLAGS', '-Wno-error=register')
+        if 'python' in self.spec.flat_dependencies() and \
+           self.spec.satisfies('@:7.99.99'):
+            self.fix_dtrace_usr_bin_path(spack_env)
```

Comment:
@chissg `setup_environment` runs as part of creating modules (not just during an install), so I think this would be better if it were moved to a build-time-only function (e.g. the way https://github.com/spack/spack/pull/7493 adds `run_before` to `fix_dtrace_usr_bin_path`). Does that seem reasonable? Or was there a reason you wanted this to be in `setup_environment` specifically?

Reply:
@scheibelp Fair enough: factored out to `@run_before('cmake')`.